source/windows/secure_channel_tls_handler.c

```diff
@@ -1812,20 +1916,199 @@ static struct aws_channel_handler *s_tls_handler_new(
 
     return NULL;
 }
+
+static DWORD getEnabledProtocols(const struct aws_tls_ctx_options *options, bool is_client_mode) {
+    DWORD grbitEnabledProtocols = 0;
+    if (is_client_mode) {
+        switch (options->minimum_tls_version) {
+            case AWS_IO_SSLv3:
+                grbitEnabledProtocols |= SP_PROT_SSL3_CLIENT;
+            case AWS_IO_TLSv1:
+                grbitEnabledProtocols |= SP_PROT_TLS1_0_CLIENT;
+            case AWS_IO_TLSv1_1:
+                grbitEnabledProtocols |= SP_PROT_TLS1_1_CLIENT;
+            case AWS_IO_TLSv1_2:
+#if defined(SP_PROT_TLS1_2_CLIENT)
+                grbitEnabledProtocols |= SP_PROT_TLS1_2_CLIENT;
+#endif
+            case AWS_IO_TLSv1_3:
+#if defined(SP_PROT_TLS1_3_CLIENT)
+                grbitEnabledProtocols |= SP_PROT_TLS1_3_CLIENT;
+#endif
+                break;
+            case AWS_IO_TLS_VER_SYS_DEFAULTS:
+                grbitEnabledProtocols = 0;
+                break;
+        }
+    } else {
+        switch (options->minimum_tls_version) {
+            case AWS_IO_SSLv3:
+                grbitEnabledProtocols |= SP_PROT_SSL3_SERVER;
+            case AWS_IO_TLSv1:
+                grbitEnabledProtocols |= SP_PROT_TLS1_0_SERVER;
+            case AWS_IO_TLSv1_1:
+                grbitEnabledProtocols |= SP_PROT_TLS1_1_SERVER;
+            case AWS_IO_TLSv1_2:
+#if defined(SP_PROT_TLS1_2_SERVER)
+                grbitEnabledProtocols |= SP_PROT_TLS1_2_SERVER;
+#endif
+            case AWS_IO_TLSv1_3:
+#if defined(SP_PROT_TLS1_3_SERVER)
+                grbitEnabledProtocols |= SP_PROT_TLS1_3_SERVER;
+#endif
+                break;
+            case AWS_IO_TLS_VER_SYS_DEFAULTS:
+                grbitEnabledProtocols = 0;
+                break;
+        }
+    }
+    return grbitEnabledProtocols;
+}
+
+static struct aws_channel_handler *s_tls_handler_new_win10_plus(

```

Comment:
I'd rather these were named based on what we're checking for semantically, ie the win32 feature and not just a windows version.